### PR TITLE
Fix add custom token 

### LINKED ISSUE #3722: wallet_watchAsset not working on mobile but works at browser extension
**Describe the bug**
wallet_watchAsset not working on mobile but works at chrome-extension
here is the code

```
try {
            // wasAdded is a boolean. Like any RPC method, an error may be thrown.
            const wasAdded = await window.ethereum.request({
              method: 'wallet_watchAsset',
              params: {
                type: 'ERC20', // Initially only supports ERC20, but eventually more!
                options: {
                  address: tokenAddress, // The address that the token is at.
                  symbol: tokenSymbol, // A ticker symbol or shorthand, up to 5 chars.
                  decimals: tokenDecimals, // The number of decimals in the token
                  image: tokenImage, // A string url of the token logo
                },
              },
            });
```

**Screenshots**
I print out the value of "wasAdded" object

photo_2022-02-09_19-58-59
IMG_3176

**To Reproduce**
_Steps to reproduce the behavior_
1. Go to 'https://flyingclub.io/switch_to_poly'
2. Click on '加入 Polygon 網路' to add polygon mainnet
3. Click on '加入 NZ 點數到 metamask' to add NZ ERC20 token on polygon

**Expected behavior**
everything works fine on browser extension

**Smartphone (please complete the following information):**
 - Device: ip12pro
 - OS: ios 15.1.1
 - App Version [e.g. 1.0.0]  v4.0.1

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### PR DESCRIPTION
**Pull request by jpcloureiro**
**Description**

_Fix payload when adding a custom token (wallet_watchAsset rpc call). Also use the token logo if provided._

[wallet_watchAsset rpc method spec](https://eips.ethereum.org/EIPS/eip-747#wallet_watchasset)

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #3722

### CODE REVIEW

**Comment by jpcloureiro** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by jpcloureiro** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by jpcloureiro** [comment]
I have read the CLA Document and I hereby sign the CLA

File: app/components/UI/AssetOverview/index.js [hunk 1]
```diff
@@ -2,10 +2,9 @@ import React, { PureComponent } from 'react';
 import { InteractionManager, StyleSheet, Text, View, TouchableOpacity } from 'react-native';
 import PropTypes from 'prop-types';
 import { swapsUtils } from '@metamask/swaps-controller';
-import AssetIcon from '../AssetIcon';
-import Identicon from '../Identicon';
 import AssetActionButton from '../AssetActionButton';
 import AppConstants from '../../../core/AppConstants';
+import TokenImage from '../../UI/TokenImage';
```
Comment (gantunesr) [inline app/components/UI/AssetOverview/index.js:7]: Can you explain why this change is necessary?

**Review by gantunesr — CHANGES_REQUESTED** [review]
Good work @jpcloureiro! Left some comments

File: app/components/UI/TokenImage/index.js [hunk 2]
```diff
@@ -18,7 +19,9 @@ const styles = StyleSheet.create({
 });
 
 export function TokenImage({ asset, containerStyle, iconStyle, tokenList }) {
-	const iconUrl = tokenList[asset?.address]?.iconUrl || tokenList[asset?.address?.toLowerCase()]?.iconUrl || '';
+	const assetImage = isUrl(asset.image) ? asset.image : null;
```
[the remark this answers is not in the document]
Comment (jpcloureiro) [reply app/components/UI/TokenImage/index.js:22]: Good catch. Somehow I assumed asset prop was declared as `isRequired`

We should also use optional when passing `asset.address` to the `<Identicon />` component (line 28)

File: app/components/UI/AssetOverview/index.js (hunk 1 above)
Comment (jpcloureiro) [reply app/components/UI/AssetOverview/index.js:7]: `TokenImage` component already implements this logic

```jsx
const iconUrl = tokenList[address]?.iconUrl || tokenList[address?.toLowerCase()]?.iconUrl || '';
return iconUrl ? <AssetIcon logo={iconUrl} /> : <Identicon address={address} />;
```

It is better to reuse that component and remove the unused imports

File: app/components/UI/TokenImage/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jpcloureiro) [reply app/components/UI/TokenImage/index.js]: addressed at https://github.com/MetaMask/metamask-mobile/pull/3942/commits/4bfbfc93ab8656ee2d440c1da7b7af93cc71e318

**Review by gantunesr — APPROVED** [review]
LGTM

**Comment by chrisleewilcox** [comment]
Getting the following error when running `yarn setup`...
**ERROR** Failed to apply patch for package @metamask/design-tokens at path
  
    node_modules/@metamask/design-tokens

  This error was caused because @metamask/design-tokens has changed since you
  made the patch file for it. This introduced conflicts with your patch,
  just like a merge conflict in Git when separate incompatible changes are
  made to the same piece of code.

  Maybe this means your patch file is no longer necessary, in which case
  hooray! Just delete it!

  Otherwise, you need to generate a new patch file.

  To generate a new one, just repeat the steps you made to generate the first
  one.

  i.e. manually make the appropriate file changes, then run 

    patch-package @metamask/design-tokens

  Info:
    Patch file: patches/@metamask+design-tokens+1.4.2.patch
    Patch was made for version: 1.4.2
    Installed version: 1.5.1

---
patch-package finished with 1 warning(s), 1 error(s).
4. Create xcconfig files...
5. Init git submodules

**Comment by wachunei** [comment]
@chrisleewilcox this was fixed here https://github.com/MetaMask/metamask-mobile/commit/b9d3f4b1b33c377508eaefab9e371bb2e62195cb, updating this branch with main will fix it.

**Review by chrisleewilcox — APPROVED** [review]
Here is a recording of the test.  Data is shown as expected and token icon is shown.

https://recordit.co/pjMnCIaTr4


LGTM

**Pull request merged**